### DISCUSSION TITLE
FreeCAD and OpenSCAD integration unavailable through FlatPak and AppImage? (Linked issues in Body)

### DISCUSSION THREAD

**Discussion by ElsterSIGNALISfanclub** [FreeCAD/FreeCAD#17889] [Discussions that will turn in to issues but need refining]
I had previously been using the Flatpak version of FreeCAD, which worked well for my needs, until I encountered an issue with using the OpenSCAD workbench -- namely, that FreeCAD could not find the actual OpenSCAD executable even when I was attempting to type it in manually or search for it in the file browser. After searching, I found an issue [here](https://github.com/FreeCAD/FreeCAD_Conda/issues/19) that seemed to exactly match with what was going on for me, but closer to the end of the discussion it is mentioned that the Flatpak version does not seem to support fixing the bugged code line in the .cfg file referenced by the [Solution](https://github.com/FreeCAD/FreeCAD_Conda/issues/19#issuecomment-731829403), and I am nowhere near advanced enough to Linux-ify and then Flatpak-ify the Solution to fix my issues.

Reading a linked issue [readable here](https://github.com/FreeCAD/FreeCAD-Bundle/issues/56) and another one [here](https://github.com/FreeCAD/FreeCAD/issues/8239), it seems the AppImage may not support OpenSCAD integration -- of note is that a specific reply mentions that the AppImage is "fixed"; [this reply](https://github.com/FreeCAD/FreeCAD/issues/8239#issuecomment-2381800316) also claims that the FlatPak version works "relatively well when using the openscad flatpak except for very specific circumstances"; the "specific circumstances" are not explicitly stated and, regardless, this **overall does not match my experience using the FlatPak over the same timeframe,** leading me to believe this may be an issue of "works on my machine."

Can we get some verification on if this issue persists in the AppImage, and if so, would a possible method of addressing this be to link the Github installation candidate as an alternate method?

For system-related issues, I am currently using Linux Mint 21.3 Cinnamon 6.0.4 on the 5.15.0-125-generic kernel.

**Comment by ElsterSIGNALISfanclub** [comment]
I am currently attempting to see if the issue persists for _myself_ as far as the AppImage goes, but other users independently verifying would be very much appreciated, as would assistance with the other issue of OpenSCAD's executable not being able to be found if possible.

Edit: While the error is *different*, the effect is the same.

**Comment by adrianinsaval** [comment]
The weekly and release candidate appimages should work with openscad now, if you are using 0.21 it probably won't work.
https://github.com/FreeCAD/FreeCAD-Bundle/releases/weekly-builds
https://github.com/FreeCAD/FreeCAD-Bundle/releases/tag/1.0rc4

If it doesn't work for you please let me know, I can't fix something I'm not aware of.

0.21 flatpak is probably not going to work that well either, but you can try the 1.0rc4 flatpak version in flathub-beta repo:
```
flatpak remote-add --if-not-exists flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo
flatpak install flathub-beta org.freecad.FreeCAD
```
I recommend trying with a clean config. There are still some things that don't directly work, if you need to import something in your openscad script it will not work by default unfortunately but there are ways to workaround it, it is not trivial to make it work by default though, fixing how this works on flatpak is on my todo list. Note that using the openscad appimage from the freecad flatpak doesn't work unless you make a wrapper script and set a transfer directory in openscad preferences.

**Reply by ElsterSIGNALISfanclub** [reply]
Will go ahead and give this a shot! Thank you.

**Comment by raviprak** [comment]
Thanks Adrian for your work in supporting FreeCAD. For the purposes of possibly helping out others, here's what I had to do on Fedora.

As of 12/16/2024, FreeCAD is only available to install via Fedora Flatpak. The fedora packaging files (https://github.com/FreeCAD/FreeCAD/tree/main/package/fedora) seem to be outdated and unused (Adrian could you please confirm?). OpenSCAD is available via RPM **or** Flatpak (you can install both versions). For OpenSCAD integration to work with FreeCAD, you **MUST** install the Flatpak version of OpenSCAD. Then, select OpenSCAD workbench, go to "Edit->Preferences->OpenSCAD". Then inside the flatpak sandbox, the OpenSCAD executable is located at `/app/bin/openscad` (NOT `/usr/bin/openscad` like it is in the RPM install).

Once you do this, OpenSCAD integration works in FreeCAD. Thanks again for all the work Adrian!
